connection/TypeDBClientImpl.java

```diff
@@ -45,12 +46,21 @@ public abstract class TypeDBClientImpl implements TypeDBClient {
     private final RequestTransmitter transmitter;
     private final TypeDBDatabaseManagerImpl databaseMgr;
     private final ConcurrentMap<ByteString, TypeDBSessionImpl> sessions;
+    protected boolean connectionValidated;
 
     protected TypeDBClientImpl(int parallelisation) {
         NamedThreadFactory threadFactory = NamedThreadFactory.create(TYPEDB_CLIENT_RPC_THREAD_NAME);
         transmitter = new RequestTransmitter(parallelisation, threadFactory);
         databaseMgr = new TypeDBDatabaseManagerImpl(this);
         sessions = new ConcurrentHashMap<>();
+        connectionValidated = false;
+    }
+
+    protected abstract void validateConnection();
+
+    @Override
+    public boolean isOpen() {
+        return !channel().isShutdown() && connectionValidated;
```

Comment:
"open" now means the channel is available and the connection has been validated

Reply:
I think these should be in different accessors now. There's currently no way to check whether the client is in a "destroyed" state (i.e. `.close()` was explicitly called), or if it's in a "not yet validated" state where it can be reopened.